cfgmgr/vxlanmgr.cpp

```diff
@@ -432,6 +452,27 @@ bool VxlanMgr::isVxlanStateOk(const std::string & vxlanName)
     return false;
 }
 
+std::string VxlanMgr::getVxlanRouterMacAddress()
+{
+    std::vector<FieldValueTuple> temp;
+
+    if (m_appSwitchTable.get(SWITCH, temp))
+    {
+        auto itr = std::find_if(
+            temp.begin(),
+            temp.end(),
+            [](const FieldValueTuple &fvt) { return fvt.first == VXLAN_ROUTER_MAC; });
+        if (itr != temp.end() && !(itr->second.empty()))
+        {
+            SWSS_LOG_DEBUG("Mac address %s is ready", itr->second.c_str());
+            return itr->second;
+        }
```

Comment:
So if the switch table is present but vxlan router mac is not present, then I wouldn't recommend retry. In this case the api can be changed to return true and no need to set mac to kernel interface. It will be auto set to switch mac.

Reply:
Done